gfx/layers/client/TiledContentClient.cpp

```diff
@@ -497,8 +497,15 @@ static already_AddRefed<TextureClient> CreateBackBufferTexture(
     aAllocator->ReportClientLost();
   }
 
+  // https://github.com/RecordReplay/backend/issues/5113
+  recordreplay::RecordReplayAssert("CreateBackBufferTexture Start");
```

Comment:
Is this assert meaningful?  The control flow will definitely flow into and out of the call to `GetTextureClient` below (unless there's a possibility of a throw, which isn't a mechanism used in gecko).  The assert _after_ the call should be equivalent to the assert before, right?

Wouldn't the assert immediately following the call to `GetTextureClient` capture this and save some recording space too?

Reply:
The concern here is more that it appears that behavior within the GetTextureClient function can vary and perform different asserts, and the assert afterwards ensures we've isolated that difference in behavior to within that function.

The assert before the call isn't nearly as important, but it helps to make sure we have consistent control flow when recording vs. replaying which is hard to be completely sure of right now because there are multiple TextureClientAllocator implementations and I'm not 100% sure which one of them is in use.

Since this code isn't super hot I'm not worried about the recording size overhead --- in general we want to prioritize getting to the root of a problem over concerns about assertion overhead, as long as we can remove the asserts once the problem has been identified.  Because assertions and other recording data are all compressed, lots of similar assertions don't have a huge impact on the final recording size.